pinot-common/src/main/java/org/apache/pinot/common/function/JsonPathMapCache.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function;
+
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.spi.cache.Cache;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * This cache for avoiding repeatedly compiling JSON path string when using jsonPathXXX transformations
+ *
+ * <p>The existing jay way LRUCache is very inefficient, It will cause
+ * a lot of unnecessary lock waits during high concurrent data ingestion,
+ * and LRU mechanism is inappropriate for Pinot bounded size of the
+ * transformation config, so we should use this simple Map cache instead
+ * of it.
+ */
+public class JsonPathMapCache implements Cache {
+  private final ConcurrentMap<String, JsonPath> _pathCache = new ConcurrentHashMap<>(128, 0.75f, 1);
```

Comment:
I don't think we need to explicitly set the concurrency level or load factor, these are the default values any way.

Reply:
I will change it.  This is the habit of my previous team!